### ISSUE TITLE
Version 1.0.1 broken under Ruby 2.7?

### ISSUE THREAD

**Issue by sidonath** [Shopify/symbol-fstring#8]
Hi,

We've updated from v1.0.0 to v1.0.1 in our Rails 6.1 (Ruby 2.7) app and the app doesn't boot anymore. Every boot attempt is failing with this stack trace:

```
$ rails r 'puts "foo"'
Traceback (most recent call last):
	30: from bin/rails:4:in `<main>'
	29: from bin/rails:4:in `require'
	28: from /var/lib/gems/2.7.0/gems/railties-6.1.3.2/lib/rails/commands.rb:18:in `<top (required)>'
	27: from /var/lib/gems/2.7.0/gems/railties-6.1.3.2/lib/rails/command.rb:50:in `invoke'
	26: from /var/lib/gems/2.7.0/gems/railties-6.1.3.2/lib/rails/command/base.rb:69:in `perform'
	25: from /var/lib/gems/2.7.0/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
	24: from /var/lib/gems/2.7.0/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
	23: from /var/lib/gems/2.7.0/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
	22: from /var/lib/gems/2.7.0/gems/railties-6.1.3.2/lib/rails/commands/runner/runner_command.rb:33:in `perform'
	21: from /var/lib/gems/2.7.0/gems/railties-6.1.3.2/lib/rails/command/actions.rb:14:in `require_application_and_environment!'
	20: from /var/lib/gems/2.7.0/gems/railties-6.1.3.2/lib/rails/command/actions.rb:22:in `require_application!'
	19: from /var/lib/gems/2.7.0/gems/railties-6.1.3.2/lib/rails/command/actions.rb:22:in `require'
	18: from /vagrant/config/application.rb:24:in `<top (required)>'
	17: from /var/lib/gems/2.7.0/gems/bundler-2.2.16/lib/bundler.rb:173:in `require'
	16: from /var/lib/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/runtime.rb:50:in `require'
	15: from /var/lib/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/runtime.rb:50:in `each'
	14: from /var/lib/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/runtime.rb:61:in `block in require'
	13: from /var/lib/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/runtime.rb:61:in `each'
	12: from /var/lib/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/runtime.rb:66:in `block (2 levels) in require'
	11: from /var/lib/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/runtime.rb:66:in `require'
	10: from /var/lib/gems/2.7.0/gems/symbol-fstring-1.0.1/lib/fstring/all.rb:6:in `<top (required)>'
	 9: from /var/lib/gems/2.7.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:332:in `require'
	 8: from /var/lib/gems/2.7.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:299:in `load_dependency'
	 7: from /var/lib/gems/2.7.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:332:in `block in require'
	 6: from /var/lib/gems/2.7.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:332:in `require'
	 5: from /var/lib/gems/2.7.0/gems/symbol-fstring-1.0.1/lib/fstring.rb:5:in `<top (required)>'
	 4: from /var/lib/gems/2.7.0/gems/symbol-fstring-1.0.1/lib/fstring.rb:6:in `<module:FString>'
	 3: from /var/lib/gems/2.7.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:332:in `require'
	 2: from /var/lib/gems/2.7.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:299:in `load_dependency'
	 1: from /var/lib/gems/2.7.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:332:in `block in require'
/var/lib/gems/2.7.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:332:in `require': cannot load such file -- fstring/fstring (LoadError)
```

Comparing the content of `ext/fstring` directories for 1.0.0 and 1.0.1 version shows some difference:

```
$ ls /var/lib/gems/2.7.0/gems/symbol-fstring-1.0.0/ext/fstring/
extconf.rb  fstring.c  fstring.o  fstring.so  Makefile
```

```
$ ls /var/lib/gems/2.7.0/gems/symbol-fstring-1.0.1/ext/fstring/
extconf.rb  fstring.c
```

The missing `.so` file is suspicious, but there's no Makefile either. Is it something in our environment that made the gem not install properly or is there a problem with the latest release?

Thank you for all the help in making apps faster!

**Comment by rafaelfranca** [comment]
The gem doesn't compile anymore if `Symbol#name` is defined https://github.com/Shopify/symbol-fstring/commit/c3c3a99d965867094c34f76c4bef912d0cb4a919.

Can you provide which exception you are getting?

**Comment by sidonath** [comment]
@rafaelfranca whoops, sorry, I missed to copy the most important line!

```
/var/lib/gems/2.7.0/gems/bootsnap-1.7.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require': cannot load such file -- fstring/fstring (LoadError)
```

I'm running this under Ruby 2.7 so `Symbol#name` doesn't exist.

```
$ pry
[1] pry(main)> :foo.name
NoMethodError: undefined method `name' for :foo:Symbol
from (pry):1:in `__pry__'
```

**Comment by rafaelfranca** [comment]
Thanks. That file should only be required `:name` is not defined in Symbol https://github.com/Shopify/symbol-fstring/blob/40ec55b643f902e38e27b1857d67d4998e871f56/lib/fstring.rb#L6.

**Comment by rafaelfranca** [comment]
Can you check if your app is defining that method somewhere else?

**Comment by casperisfine** [comment]
Uh, weird. I'll try to take a look. Thanks for pinging me, I forgot to watch the repository.

**Comment by casperisfine** [comment]
Yes, that was it. This `unless` in the gemspec is to total brainfart 🤦 .

I released a 1.0.2, that should fix your problem @sidonath

**Comment by sidonath** [comment]
@casperisfine it does, thank you! 🎉
